src/camel_snake_kebab/internals/misc.cljc

```diff
@@ -5,7 +5,8 @@
 (defn convert-case [first-fn rest-fn sep s & {:keys [separator]
                                               :or   {separator generic-separator}}]
   (let [[first & rest] (split separator s)]
-    (join sep (cons (first-fn first) (map rest-fn rest)))))
+    (join sep (cons (if first (first-fn first) "") 
```

Comment:
What do you think about switching the implementation to `(if-let [[first & rest] (seq (split separator s))]
  (join sep (cons (first-fn first) (map rest-fn rest)))
  "")` instead?